text/0000-edition-2021.md

```diff
@@ -0,0 +1,226 @@
+
+### Uniform behavior across editions
+
+Persuant to the goal of having Rust feel like "one language", we generally prefer uniform behavior across all editions of Rust, so long as it can be achieved without compromising other design goals. This means for example that if we add a new feature that is backwards compatible, it should be made available in all editions. Similarly, if we deprecate a pattern that we think is problematic, it should be deprecated across all editions.
+
+Having uniform behavior across editions has several benefits:
+
+* Reducing technical debt: the compiler is simpler if it has to consider fewer cases.
+* Minimizing cognitive overload: we want to avoid users having to think too much about what edition they are in. We would prefer that all Rust code feels the same, no matter what edition it is using, although we are willing to compromise on this principle if the benefits are large.
+    * Even when changes alter core parts of Rust, we are often able to introduce parts of those changes across all editions, which helps us to achieve more uniformity. For example, the `crate::foo::bar` paths introduced in Rust 2018 are also available on Rust 2015, since that syntax had no meaning before.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Edition cadence not specified
+
+This RFC explicitly does not attempt to specify the cadence for releasing editions. Historically, they have occurred every 3 years, if one considers Rust 1.0 (released in 2015) to be the "original edition". Our expectation is that after Rust 2021 is shipped, we will use the experiences from Rust 2018 and Rust 2021 to review how editions are working and potentially establish a cadence for future editions (or perhaps make other changes to the edition system). 
+
+There are various constraints on the timing of editions that have been identified over time:
+
+* Even if it's automated, migrating to a new edition can require a significant investment of time, particularly for production users with many crates. We don't wish to release Editions at a rate that makes it hard for such users to keep up.
+* At the same time, we don't wish to wait too long in between editions. We want to maintain the spirit of Rust's train model, so that folks don't feel a "rush" to get work done by any particular deadline.
+* Having a generally known time when editions will be released is helpful in planning feature development.
+* We do not wish to release an empty edition. Sometimes there simply won't be any backwards incompatible changes and thus no reason to issue an edition.
+
+## Definition: migration
+
+**Migrations** are the "breaking changes" introduced by the edition, except of course that since editions are opt-in, no code actually breaks. 
```

Comment:
```suggestion
**Migrations** are the "breaking changes" introduced by the edition; of course, since editions are opt-in, code cannot break until you ask it to, thus upholding our stability policy.
```